src/badguy/zeekling.cpp

```diff
@@ -119,57 +121,34 @@ Zeekling::collision_solid(const CollisionHit& hit)
   }
 }
 
-/** Linear prediction of player and badguy positions to decide if we should enter the DIVING state. */
 bool
 Zeekling::should_we_dive()
 {
-  if (m_frozen)
-    return false;
-
-  const auto player = get_nearest_player();
-  if (player && last_player && (player == last_player)) {
-
-    // Get positions and calculate movement.
-    const Vector& player_pos = player->get_pos();
-    const Vector player_mov = (player_pos - last_player_pos);
-    const Vector self_pos = m_col.m_bbox.p1();
-    const Vector self_mov = (self_pos - last_self_pos);
-
-    // New vertical speed to test with.
-    float vy = 2*fabsf(self_mov.x);
-
-    // Do not dive if we are not above the player.
-    float height = player_pos.y - self_pos.y;
-    if (height <= 0) return false;
+  if (m_frozen) return false;
 
-    // Do not dive if we are too far above the player.
-    if (height > 512) return false;
+  // Left/rightmost point of the hitbox.
+  Vector eye;
+  const Rectf& bbox = get_bbox().grown(1.f);
+  eye = bbox.get_middle();
+  eye.x = m_dir == Direction::LEFT ? bbox.get_left() : bbox.get_right();
 
-    // Do not dive if we would not descend faster than the player.
-    float relSpeed = vy - player_mov.y;
-    if (relSpeed <= 0) return false;
+  const Vector& plrmid = get_nearest_player()->get_bbox().get_middle();
```

Comment:
That name is not very verbose.

Reply:
player middle. i thinks its pretty ok. what are your suggestions

Reply:
`player_middle`, `player_center`?

Reply:
sure